FirebaseFunctions/Sources/Internal/FunctionsSerializer.swift

```diff
@@ -106,73 +91,70 @@ final class FunctionsSerializer: Sendable {
-    // Recover the underlying type of the number, using the method described here:
-    // http://stackoverflow.com/questions/2518761/get-type-of-nsnumber
-    let cType = number.objCType
-
-    // Type Encoding values taken from
-    // https://developer.apple.com/library/mac/documentation/Cocoa/Conceptual/ObjCRuntimeGuide/
-    // Articles/ocrtTypeEncodings.html
-    switch cType[0] {
-    case CChar("q".utf8.first!):
-      // "long long" might be larger than JS supports, so make it a string.
-      return ["@type": Constants.longType, "value": "\(number)"] as AnyObject
-
-    case CChar("Q".utf8.first!):
-      // "unsigned long long" might be larger than JS supports, so make it a string.
-      return ["@type": Constants.unsignedLongType,
-              "value": "\(number)"] as AnyObject
-
-    case CChar("i".utf8.first!),
-         CChar("s".utf8.first!),
-         CChar("l".utf8.first!),
-         CChar("I".utf8.first!),
-         CChar("S".utf8.first!):
-      // If it"s an integer that isn"t too long, so just use the number.
-      return number
-
-    case CChar("f".utf8.first!), CChar("d".utf8.first!):
-      // It"s a float/double that"s not too large.
-      return number
-
-    case CChar("B".utf8.first!), CChar("c".utf8.first!), CChar("C".utf8.first!):
-      // Boolean values are weird.
-      //
-      // On arm64, objCType of a BOOL-valued NSNumber will be "c", even though @encode(BOOL)
-      // returns "B". "c" is the same as @encode(signed char). Unfortunately this means that
-      // legitimate usage of signed chars is impossible, but this should be rare.
-      //
-      // Just return Boolean values as-is.
-      return number
-
+  private func wrapNumberIfNeeded(_ number: NSNumber) -> Any {
+    switch String(cString: number.objCType) {
+    case "q":
+      // "long long" might be larger than JS supports, so make it a string:
+      return WrappedNumber(type: .long, value: "\(number)").encoded
+    case "Q":
+      // "unsigned long long" might be larger than JS supports, so make it a string:
+      return WrappedNumber(type: .unsignedLong, value: "\(number)").encoded
     default:
-      // All documented codes should be handled above, so this shouldn"t happen.
-      throw Error.unknownNumberType(charValue: String(cType[0]), number: number)
+      // All other types should fit JS limits, so return the number as is:
+      return number
     }
   }
 
-  private func decodeWrappedType(_ type: String, _ value: String) throws -> AnyObject? {
-    switch type {
-    case Constants.longType:
-      let formatter = NumberFormatter()
-      guard let n = formatter.number(from: value) else {
-        throw Error.invalidValueForType(value: value, requestedType: type)
+  private func unwrapNumber(_ wrapped: WrappedNumber) throws(Error) -> any Numeric {
+    switch wrapped.type {
+    case .long:
+      guard let n = Int(wrapped.value) else {
+        throw .failedToParseWrappedNumber(wrapped)
```

Comment:
```suggestion
        throw .invalidValueForType(value: wrapped.value, requestedType: wrapped.type.rawValue)
```

Reply:
b99bf01ef3b9fd4440d6bab2d62d3472b075db95

Reply:
I changed the case name, though, to make it clear this error is specifically about wrapped numbers, not arbitrary decoding scenarios.